pkg/api/v1/types.go

```diff
@@ -289,3 +292,63 @@ func NewVMReferenceFromName(name string) *VM {
 	vm.SetGroupVersionKind(schema.GroupVersionKind{Group: GroupVersion.Group, Kind: "VM", Version: GroupVersion.Version})
 	return vm
 }
+
+//TODO validate that this is correct
+func NewMigrationReferenceFromName(name string) *Migration {
+	migration := &Migration{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      name,
+			Namespace: kubeapi.NamespaceDefault,
+			SelfLink:  fmt.Sprintf("/apis/%s/namespaces/%s/%s", GroupVersion.String(), kubeapi.NamespaceDefault, name),
+		},
+	}
+	migration.SetGroupVersionKind(schema.GroupVersionKind{Group: GroupVersion.Group, Kind: "VM", Version: GroupVersion.Version})
+	return migration
+}
+
+type Migration struct {
+	metav1.TypeMeta `json:",inline"`
+	ObjectMeta      v1.ObjectMeta `json:"metadata,omitempty"`
+	Spec            MigrationSpec `json:"spec,omitempty" valid:"required"`
+	Status          string        `json:"message,omitempty"`
+}
+
+// VMSpec is a description of a VM
+type MigrationSpec struct {
+	MigratingVMName         string            `json:"nodeName,omitempty"`
```

Comment:
Whenever in kubernetesis selecting objects, which are manipulated (in our case VMs), the objects have a `selector` sub-field in the `spec`.
The rest of the fiels inside the `spec` are in our case also something like selectors, but I would prefer to call them additional `constraints`.

See for example the deployment: https://kubernetes.io/docs/user-guide/deployments/#selector

I don't insist on nesting it for now. Let's see what we want to be able to do with the migration object in the future.